src/components/common/TypeaheadSelect/utils.ts

```diff
@@ -0,0 +1,94 @@
+import { t } from '@utils/i18n';
+
+import { PLACEHOLDER_VALUES } from './constants';
+import type { FilterOptionsConfig, TypeaheadSelectOption } from './types';
+
+/**
+ * Creates a "create new option" item for the dropdown
+ */
+const createNewOption = (
+  inputValue: string,
+  createMessage: string | ((value: string) => string),
+): TypeaheadSelectOption => ({
+  content: typeof createMessage === 'function' ? createMessage(inputValue) : createMessage,
+  optionProps: { className: 'pf-m-action' },
+  value: inputValue,
+});
+
+/**
+ * Creates a disabled "no results found" option
+ */
+const createNoResultsOption = (
+  inputValue: string,
+  noResultsMessage: string | ((filter: string) => string),
+): TypeaheadSelectOption => ({
+  content: typeof noResultsMessage === 'function' ? noResultsMessage(inputValue) : noResultsMessage,
+  optionProps: { isDisabled: true },
+  value: PLACEHOLDER_VALUES.NO_RESULTS,
+});
+
+/**
+ * Checks if a create option should be added to the filtered results
+ */
+const shouldAddCreateOption = (
+  inputValue: string,
+  filteredOptions: TypeaheadSelectOption[],
+  isCreatable: boolean,
+): boolean =>
+  isCreatable &&
+  Boolean(inputValue.trim()) &&
+  !filteredOptions.some(
```

Comment:
I'd add optional chaining in case filterOptions is not defined to avoid crashing (waiting for data or something similar)